lambdas/handlers/create_document_reference_handler.py

```diff
@@ -19,23 +20,42 @@
 def lambda_handler(event, context):
     logger.info("Starting document reference creation process")
 
+    try:
+        document_type = (event["queryStringParameters"]["documentType"]).upper()
+        if document_type in SupportedDocumentTypes.list_names():
```

Comment:
I personally find it useful to give my enums a method where you can pass in a value and they return the appropriate Enum for that value, or None if there's not one. Might make it a bit more readable if you do:

```
document_type = SupportedDocumentTypes.from_value((event["queryStringParameters"]["documentType"]).upper())
if document_type is None:
    raise Exception
logger.info("Provided document is supported")
```

Then you'll have doc type as the enum itself so you don't have to check against SupportedDocumentTypes.LG.name below, you can drop the ".name"